### PR TITLE
support running active files with filenames containing parenthesis

### LINKED ISSUE #114556: Cannot "Run Active File" if the filename has parentheses
Issue Type: <b>Bug</b>

This is quite similar to issue #90415.
If the current filename has parentheses, such as *MyCode().ahk*, running the file via Code > Terminal > Run Active File will return an error, whether the default shell is CMD or PowerShell. For example, running a file located at 
`c:\Code\PowerAssistantMini().onmessage.ahk` would give us:

```
PS C:\Users\Name> c:\Code\PowerAssistantMini().onmessage.ahk
At line:1 char:28
+ c:\Code\PowerAssistantMini().onmessage.ahk
+                                                            ~
An expression was expected after '('.
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : ExpectedExpression`
```
Naming files with parentheses might be bad practice but, similar to the "spaces in the filename" issue (#90415), we can remedy the problem by adding "&" and enclosing the directory with quotes. Using the example directory above, we instead write:
`& "c:\Code\PowerAssistantMini().onmessage.ahk"`

In the terminal it would look like this:
```
PS C:\Users\Name> & "c:\Code\PowerAssistantMini().onmessage.ahk"
```
Visual Studio Code automatically does this when it detects spaces in the filename. I should also note that this is not limited to .ahk files. Files that have parentheses in their names will also run into this issue.

VS Code version: Code 1.52.1 (ea3859d4ba2f3e577a159bc91e3074c5d85c0523, 2020-12-16T16:34:46.910Z)
OS version: Windows_NT x64 10.0.19042

<details>
<summary>System Info</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-userscript|adp|1.3.2
xml|Dot|2.5.1
vscode-html-css|ecm|1.9.1
gc-excelviewer|Gra|3.0.40
rainbow-csv|mec|1.8.0
python|ms-|2020.12.424452561
jupyter|ms-|2020.12.414227025
sublime-keybindings|ms-|4.0.7
vscode-autohotkey|sle|0.2.2

</details>

### PR DESCRIPTION
**Pull request by meganrogge**
This PR fixes #114556

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 1]
```diff
@@ -649,6 +650,11 @@ export class TerminalService implements ITerminalService {
 				return;
 			}
 
+			if (hasParens) {
+				c(`& '${originalPath}'`);
+				return;
+			}
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts:656]: I believe `& '...'` is a powershell only thing?

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 2]
```diff
@@ -651,8 +651,13 @@ export class TerminalService implements ITerminalService {
...
-				c(`& '${originalPath}'`);
-				return;
+				if (isPowerShell) {
+					c(`& '${originalPath}'`);
+					return;
+				} else {
+					c(`'${originalPath}'`);
+					return;
+				}
 			}
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts]: Does `'path'` work on cmd and git bash or is it even needed at all? Also should this be windows specific?

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts]: I tried it and it does work on those.

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts]: in the issue, the person describes using both powershell and the cmd prompt, so I thought they should both work?

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts]: interestingly, the run active file in terminal online seems to work in git bash when the path is wrapped in quotes. and currently that's not happening other than if there are parenthesis.

but the opposite is true for cmd prompt - when there are parenthesis, it doesn't work.

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts]: > Also should this be windows specific?

can't powershell be used on non-windows devices?

**Review by Tyriar — APPROVED** [review]

**Pull request merged**